Add cleanup of all old clusters

YiqinZhang opened the pull request:
Update the osde2e cleanup command to ensure that all clusters older than given duration are deleted.


[SDCICD-1240](https://issues.redhat.com/browse/SDCICD-1240)

YiqinZhang commented:
/retest

jbpratt commented on go.mod:
```diff
@@ -21,7 +21,7 @@ require (
 	github.com/onsi/ginkgo/v2 v2.17.1
 	github.com/onsi/gomega v1.32.0
-	github.com/openshift-online/ocm-sdk-go v0.1.410
+	github.com/openshift-online/ocm-sdk-go v0.1.378
```
I don't think these changes are necessary

jbpratt commented on cmd/osde2e/cleanup/cmd.go:
```diff
@@ -93,29 +102,39 @@ func run(cmd *cobra.Command, argv []string) error {
+						if err := provider.DeleteCluster(cluster.ID()); err != nil {
+							log.Printf("Error deleting cluster: %s", err.Error())
+						} else {
+							fmt.Printf("Cluster %s Deleted\n", cluster.ID())
```
```suggestion
							log.Printf("Cluster %s Deleted\n", cluster.ID())
```

jbpratt commented:
@YiqinZhang could you paste the output from when you run the command?

YiqinZhang commented:
> @YiqinZhang could you paste the output from when you run the command?

When I dry ran the command ` ./out/osde2e cleanup --clusters=true --dry-run --older-than=24h --configs=rosa,stage`, the output was
```2024/04/11 18:18:35 configs.go:24: Will load config rosa
2024/04/11 18:18:35 configs.go:24: Will load config stage
2024/04/11 18:18:35 configs.go:24: Will load config sts
Cluster will be deleted: 26hq23dvcrsueamlet9qmogg26g9cr20 (created 2023-09-280)
Cluster will be deleted: 27m20jfd6gtitlpqcedoru4l0nior9al (created 2023-11-220)
Cluster will be deleted: 29dve9i76uktid6nvbgu7joc8f8iv68t (created 2024-02-150)
Cluster will be deleted: 2adqcm7qerutg8vuhqdk8ha0hm56mmo8 (created 2024-04-40)
```

If running without `--dry-run`, the output is expected to be:  
```
2024/04/11 18:18:35 configs.go:24: Will load config rosa
2024/04/11 18:18:35 configs.go:24: Will load config stage
2024/04/11 18:18:35 configs.go:24: Will load config sts
Cluster will be deleted: 26hq23dvcrsueamlet9qmogg26g9cr20 (created 2023-09-280)
Deleted
Cluster will be deleted: 27m20jfd6gtitlpqcedoru4l0nior9al (created 2023-11-220)
Deleted
Cluster will be deleted: 29dve9i76uktid6nvbgu7joc8f8iv68t (created 2024-02-150)
Deleted
Cluster will be deleted: 2adqcm7qerutg8vuhqdk8ha0hm56mmo8 (created 2024-04-40)
Deleted
```
